src/types.ts

```diff
@@ -2079,6 +2104,22 @@ export class ZodArray<
     return this._def.type;
   }
 
+  _arrayUnique(array: unknown[], identifier?: ArrayUniqueIdentifier) {
```

Comment:
Can you preferably take an option as input, basically asking if the user wants to display unique elements. Finding duplicates itself is an expensive operation especially if objects are involved. So it is better that we don't display unique elements by default.